cpp/src/arrow/dataset/partition_test.cc

```diff
@@ -61,12 +61,12 @@ class TestPartitioning : public ::testing::Test {
     // formatted partition expressions are bound to the schema of the dataset being
```

Comment:
This method should be updated to:

```
void AssertFormat(compute::Expression expr, const std::string& expected_directory, const std::string& expected_prefix  = "")
```

Then it should check both `first` and `second`.  Finally, we should have at least one test case for filename partitioning that uses `AssertFormat`.

Reply:
Added test for FilenamePartitioningFormat.